File: lang/ruby/src/fun/vac/algorithms/number_theory/Parity.rb
module Parity

    def self.moduloIsEven(n)
        (n % 2).zero?
    end

    def self.moduloIsOdd(n)
        n % 2 != 0
    end

    def self.bitwiseIsEven(n)
        (n & 1).zero?
    end

    def self.bitwiseIsOdd(n)
        (n & 1) != 0
    end
end

File: lang/java/src/fun/vac/algorithms/number_theory/Parity.java
package fun.vac.algorithms.number_theory;

public final class Parity {

    private Parity() {}

    public static boolean moduloIsEven(long n) {
        return n % 2 == 0;
    }

    public static boolean moduloIsOdd(long n) {
        return n % 2 != 0;
    }

    public static boolean bitwiseIsEven(long n) {
        return (n & 1) == 0;
    }

    public static boolean bitwiseIsOdd(long n) {
        return (n & 1) != 0;
    }
}
